cli/lsp/tsc.rs

```diff
@@ -3165,41 +3514,30 @@ struct Response {
   data: Value,
 }
 
-struct State {
-  last_id: usize,
-  performance: Arc<Performance>,
-  response: Option<Response>,
-  state_snapshot: Arc<StateSnapshot>,
-  normalized_specifiers: HashMap<String, ModuleSpecifier>,
-  denormalized_specifiers: HashMap<ModuleSpecifier, String>,
-  token: CancellationToken,
+#[derive(Debug, Default)]
+pub struct TscSpecifierMap {
+  normalized_specifiers: Mutex<HashMap<String, ModuleSpecifier>>,
+  denormalized_specifiers: Mutex<HashMap<ModuleSpecifier, String>>,
```

Comment:
Multiple locks on a struct is a little dangerous even though it's used fine here. We could make this code safer by doing one of the following:

1. Put the entire state behind a single mutex.
2. Use [dashmap](https://docs.rs/dashmap/5.5.3/dashmap/struct.DashMap.html) for the maps, which already is a transitive dependency of ours. That buries the Mutex within it and creates an API that doesn't allow for mistakes.